### ISSUE TITLE
Checkbox and radio buttons are not perceivable or operable.

### ISSUE THREAD

**Issue by fuzzbomb** [thememachine/eleven#161]
Sound the a11y klaxon! This one is a major barrier.  Failure of 2 WCAG success criteria at level A:
 - [SC 2.1.1 Keyboard](https://www.w3.org/TR/UNDERSTANDING-WCAG20/keyboard-operation-keyboard-operable.html)
 - [SC 2.4.3 Focus Order](https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-focus-order.html)

## Problem.
Radio buttons and checkboxes are not keyboard focusable.  This makes them completely unusable by several groups of users:

- Screen reader users with little or no vision will be unlikely to know that these inputs are event present at all.  A screen reader user will not find these by using the TAB key, or by using any forms tools provided by various screen readers (e.g. features such as skip-to-next-form-input, or list-form-inputs).
  - A screen reader power user _might_ be able to find the label elements in normal reading mode, and use a "force mouse click on current text" feature, but this would be a user with exceptional patience, so it's not a _viable_ workaround. Definite WCAG level A fail.
- For sighted keyboard users (with or without a screen reader), these inputs cannot be focused using the TAB key, despite being able to see them clearly.
  - A workaround exists using a "mouse keys" feature (num pad as 8 arrows, emulates mouse events). Some users may be confident with mouse keys; for others it will be a PITA.  Some users may not have a number pad, or maybe require it for some other macros. Definite WCAG level A fail.
- I have no idea how this will perform using switch access scanning (a.k.a. switch control).  There is a cross-hair positioning mouse-click feature in macOS Switch Control.

## Cause of problem.
The `input[type=checkbox]` and `input[type=radio]`  are styled with `visibility: hidden`.  This has the effect of:
 - taking them out of the tabbing order in the browser, and
 - taking them out of the accessibility tree (a DOM subset passed to the host OS accessibility APIs, so they don't get passed to screen readers and other assistive tech).

## Steps to reproduce.
I picked some simple forms which demonstrate this well, which have visble focus on elements before and after the checkbox/radio inputs.

- Checkboxes - visit `admin/structure/menu/link/filter.tips_all/edit`.
  - Press TAB until the "Add to default shortcuts" tooltip appears.
  - Press TAB once.  The "Compose tips" link now has focus, although it doesn't have a visible focus style (another bug...)  You might see the URL (`filter/tips`) pop up in the status bar of your browser.
  - Press TAB once.  The "Parent link" select element now has focus.  The checkboxes didn't receive focus.
  - Press shit-TAB twice, you'll end up back at the add-shortcut control.
- Radio buttons - visit `admin/config/development/logging`.
  - Press TAB until the "Add to default shortcuts" tooltip appears.
  - Press TAB once more. The "Database log messages to keep" now has focus.  The radios didn't receive focus.
  - Press shift-TAB once, now focus is back on the add-shortcut control.

## Remedy.
1. Do not use `visibility: hidden` or `display: none` to hide inputs.  Find another way to style these form elements. Maybe opacity property or a visually-hidden class.
  I have a test page where you can see the effects of different ways of hiding controls with CSS.  Play with [Hidden elements - focus test](https://fuzzbomb.github.io/accessibility-demos/visually-hidden-focus-test.html) using TAB, shift-TAB, and Enter keys.
2. Remove the duplicate `<label>` element.  If you must have a fake checkbox, a span inside the existing label will do, and can still be targeted with `input:checked + label > span` (after the input itself has been made operable in the previous step).

**Comment by mortendk** [comment]
Auuuuch! ;)

**Comment by fuzzbomb** [comment]
Sweet! They are working well now for sighted keyboard and screen reader operation.  They also scale well when using browser zoom.

**Comment by fuzzbomb** [comment]
Just a note to self that the fixes were in commit [dc41277](https://github.com/thememachine/eleven/commit/dc41277b04e09d201798f463ccd7de3586d0a1dd).  I sometimes use this issue as an example.
